src/nvim/terminal.c

```diff
@@ -348,11 +348,17 @@ void terminal_close(Terminal **termpp, int status)
 
   if (buf && !is_autocmd_blocked()) {
     save_v_event_T save_v_event;
+    aco_save_T aco;
+
     dict_T *dict = get_v_event(&save_v_event);
     tv_dict_add_nr(dict, S_LEN("status"), status);
     tv_dict_set_keys_readonly(dict);
+    aucmd_prepbuf(&aco, buf);
```

Comment:
The `buf->b_locked++` guard from https://github.com/neovim/neovim/pull/19222 fixes #10386 except for the edge case where altbuf is the first default empty buffer. I think the remaining work needed is to figure out why `do_buffer` calls `set_curbuf` with the wrong arg.

So I don't think we need/want the `aucmd_prepbuf` dance here.